### PR TITLE
PROGRAMMES-6260 & PROGRAMMES-6261 - Speech (and Chapter) Segments

### PR DESCRIPTION
**Pull request by felipeparaujo**
We're removing playout from Episode pages, and future systems won't support deep linking, so Chapter and Speech segments are the same.

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/Ds2013/Presenters/Section/Segments/SegmentItem/GroupPresenter.php [hunk 1]
```diff
@@ -8,14 +8,10 @@ class GroupPresenter extends AbstractSegmentItemPresenter
     /** @var AbstractSegmentItemPresenter[] */
     private $presenters;
 
-    /** @var string */
-    private $title;
-
     public function __construct(string $title, array $presenters, array $options)
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Section/Segments/SegmentItem/GroupPresenter.php]: Don't you need to change this if you're passing in a SegmentEvent now?

File: src/Ds2013/Presenters/Section/Segments/SegmentItem/SpeechPresenter.php [hunk 2]
```diff
@@ -3,26 +3,26 @@
...
+    public function getDuration(): string
     {
-        return $this->segment->getTitle();
+        $format = 'i:s';
+
+        if ($this->segmentEvent->getSegment()->getDuration() >= 3600) {
+            $format = 'h:i:s';
+        }
+
+        return gmdate($format, $this->segmentEvent->getSegment()->getDuration());
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Section/Segments/SegmentItem/SpeechPresenter.php]: That's going to fail if the programme is >24h long. There are things that long in the DB, although they're mostly "barkers" which won't have segments. I can't think of a "nice" way to do this either though tbh.

File: src/Ds2013/Presenters/Section/Segments/SegmentsListPresenter.php [hunk 3]
```diff
@@ -251,15 +250,9 @@ private function createSegmentItem(array $segmentEvents, int $start): AbstractSe
...
-            } elseif ($segmentEvent->isChapter() &&
-                $segmentEvent->getOffset() &&
-                ($this->context->isStreamable() || $this->context->isDownloadable()) &&
-                !$this->isLive
-            ) {
-                $presenters[] = new ChapterPresenter($segmentEvent->getSegment(), $options);
+                $presenters[] = new MusicPresenter($segmentEvent, $options);
             } else {
-                $presenters[] = new SpeechPresenter($segmentEvent->getSegment(), $options);
+                $presenters[] = new SpeechPresenter($segmentEvent, $options);
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Section/Segments/SegmentsListPresenter.php]: Shouldn't we still be checking segment type here, what if new ones get added or something? We could just ignore the ones we don't understand. Or am I missing something?

File: src/Ds2013/Presenters/Section/Segments/SegmentItem/speech.html.twig [hunk 4]
```diff
@@ -1,6 +1,13 @@
-<li class="segments-list__item segments-list__item--speech {{ speech.getOption('moreless_class') }}">
-    <{{ speech.getOption('h_tag') }} class="gamma no-margin">
-        {{ speech.getTitle() }}
-    </{{ speech.getOption('h_tag') }}>
-    <p>Speech Segment</p>
+<li class="segments-list__item segments-list__item--music {{ speech.getOption('moreless_class') }}">
+    <div class="segment text--prose">
+        <div class="segment__content">
+            {{ speech.getPid() }}
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Section/Segments/SegmentItem/speech.html.twig]: This method doesn't exist, also why?

Comment (felipeparaujo) [reply src/Ds2013/Presenters/Section/Segments/SegmentItem/speech.html.twig]: Just debugging stuff that got leftover, I'll delete it

File: src/Ds2013/Presenters/Section/Segments/SegmentsListPresenter.php (hunk 3 above)
Comment (felipeparaujo) [reply src/Ds2013/Presenters/Section/Segments/SegmentsListPresenter.php]: V2 defaults to speech when it doesn't recognize the type of the segment. I thought we should be doing the same here.

Comment (felipeparaujo) [reply src/Ds2013/Presenters/Section/Segments/SegmentsListPresenter.php]: https://github.com/bbc/programmes/blob/develop/webapp/php/lib/BBC/Programmes/View/SegmentItems.php#L263-L279

File: src/Ds2013/Presenters/Section/Segments/SegmentItem/GroupPresenter.php (hunk 1 above)
Comment (felipeparaujo) [reply src/Ds2013/Presenters/Section/Segments/SegmentItem/GroupPresenter.php]: Yes.

File: src/Ds2013/Presenters/Section/Segments/SegmentItem/SpeechPresenter.php (hunk 2 above)
Comment (felipeparaujo) [reply src/Ds2013/Presenters/Section/Segments/SegmentItem/SpeechPresenter.php]: https://github.com/bbc/programmes-frontend/blob/develop/src/Ds2013/Presenters/Domain/CoreEntity/Programme/SubPresenters/ProgrammeBodyPresenter.php#L53-L69

The programme object does this. How about extracting it to a helper?

Comment (BjornTwachtmann) [reply src/Ds2013/Presenters/Section/Segments/SegmentItem/SpeechPresenter.php]: That seems like a good approach, yeah

File: src/Ds2013/Presenters/Section/Segments/SegmentsListPresenter.php (hunk 3 above)
Comment (BjornTwachtmann) [reply src/Ds2013/Presenters/Section/Segments/SegmentsListPresenter.php]: Fair enough, leave it this way for now. I think there's a ticket somewhere for a new type anyway.

**Pull request merged**